src/main/java/com/thebois/models/world/structures/House.java

```diff
@@ -10,20 +14,32 @@
     /**
      * Creates a house structure at a given position in the world.
      *
-     * @param posX Position in X-axis
-     * @param posY Position in Y-axis
+     * @param posX Position in X-axis.
+     * @param posY Position in Y-axis.
      */
     public House(final int posX, final int posY) {
-        super(posX, posY, StructureType.HOUSE);
+        this(new Position(posX, posY));
     }
 
     /**
      * Creates a house structure at a given position in the world.
      *
-     * @param position The position of the house
+     * @param position The position of the house.
      */
     public House(final Position position) {
-        super(position, StructureType.HOUSE);
+        super(position, StructureType.HOUSE, generateNeededItemsInventory());
+    }
+
+    private static IInventory generateNeededItemsInventory() {
+        final IInventory neededItems = new Inventory();
+
+        final int numberOfItems = 10;
+        for (int i = 0; i < numberOfItems; i++) {
+            neededItems.add(new Log());
+            neededItems.add(new Rock());
+        }
+
+        return neededItems;
```

Comment:
Feels like a map is of better use in this scenario, would be something that would be like `RocksNeeded -> 10`